lib/datadog/core/utils/safe_dup.rb

```diff
@@ -1,40 +1,39 @@
+require_relative '../backport'
+
 module Datadog
   module Core
     module Utils
       # Helper methods for safer dup
       module SafeDup
-        if RUBY_VERSION < '2.2' # nil.dup only fails in Ruby 2.1
-          # Ensures #initialize can call nil.dup safely
-          module RefineNil
-            refine NilClass do
-              def dup
-                self
-              end
-            end
-          end
-
-          using RefineNil
-        end
-
         # String#+@ was introduced in Ruby 2.3
         if String.method_defined?(:+@) && String.method_defined?(:-@)
           def self.frozen_or_dup(v)
-            # If the string is not frozen, the +(-v) will:
-            # - first create a frozen deduplicated copy with -v
-            # - then it will dup it more efficiently with +v
-            v.frozen? ? v : +(-v)
+            case v
+            when String
```

Comment:
Can you capture here why `String` is handled separately from other types?

Reply:
Done https://github.com/DataDog/dd-trace-rb/pull/2960/commits/f4429a6d9501cc9e20985d92cdd71a0b30a40fa9

Reply:
Minor: Isn't `case v .... when String ... else ...` the same as `if String === v ... else`? (Same applies below as well)